grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -557,6 +559,26 @@ class StockMovementService {
         }
 
         switch(stepNumber) {
+            case "2":
+                if (requisition && requisition.sourceType == RequisitionSourceType.ELECTRONIC && requisition.requisitionTemplate) {
+                    stockMovementItems = stockMovementItems.collect {stockMovementItem ->
+                        def quantityOnHand = ProductAvailability.findAllByProductAndLocation(stockMovementItem.product, requisition.destination).sum { it.quantityOnHand }
```

Comment:
@DominikaHatala please add a new method in ProductAvailabilityService for this purpose.
It can be a:
`Integer getQuantityOnHand(Product product, Location location) {...}`